### PR TITLE
libcoap: Add new package

### PR DESCRIPTION
**Pull request by toxxin**
Signed-off-by: Anton Glukhov <anton.a.glukhov@gmail.com>

### CODE REVIEW

File: libs/libcoap/Makefile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/obgm/libcoap
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)
+PKG_SOURCE_VERSION:=683337890e6fbab1eaa0ef0f2038194fb62b7fa2
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_SOURCE_VERSION).tar.bz2
+
+PKG_MAINTAINER:=Anton Glukhov <anton.a.glukhov@gmail.com>
+PKG_LICENSE:=GPL-2.0+
+PKG_LICENSE_FILE:=LICENSE.GPL
```
Comment (karlp) [inline libs/libcoap/Makefile]: (GPL-2.0+ or BSD-2-Clause)

File: libs/libcoap/patches/0001-manpage.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
+--- a/examples/Makefile.am
++++ b/examples/Makefile.am
+@@ -23,18 +23,6 @@
+ coap_rd_SOURCES = coap-rd.c
+ coap_rd_LDADD = @top_builddir@/.libs/libcoap-$(LIBCOAP_API_VERSION).la
+ 
+-# building the manpages
```
Comment (karlp) [inline libs/libcoap/patches/0001-manpage.patch]: Man, I thought this was wrong, but it seems --disable-documentation is just busted upstream, seems like it only disables the checks on building them.  That's vile, please file a bug upstream for that.

Comment (toxxin) [inline libs/libcoap/patches/0001-manpage.patch]: Yeah, --disable-documentation doesn't eliminate man5 generation. Let's handle it by patch for a while, and I'll try to figure out with it and make patch for upstream source soon.

**Comment by karlp** [comment]
it's been fixed upstream already for us :)

**Comment by toxxin** [comment]
Applied last changes [https://github.com/obgm/libcoap/issues/11](https://github.com/obgm/libcoap/issues/11)

File: libs/libcoap/Makefile [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+define Package/libcoap/description
+  Constrained Application Protocol(RFC 7252) library
+endef
+
+TARGET_CFLAGS += $(FPIC)
+
+CONFIGURE_ARGS += \
+	--enable-shared \
+	--enable-static \
```
Comment (karlp) [inline libs/libcoap/Makefile]: So, why are you enabling static actually?

File: libs/libcoap/Makefile [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+define Package/libcoap
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE:=CoAP(RFC 7252) library
+  URL:=http://libcoap.net/
+endef
+
+define Package/libcoap/description
+  Constrained Application Protocol(RFC 7252) library
```
Comment (karlp) [inline libs/libcoap/Makefile]: spaces before the (RFC... in both lines probably wouldn't hurt :)

File: libs/libcoap/Makefile (hunk 3 above)
Comment (toxxin) [inline libs/libcoap/Makefile]: Static and dynamic are enabled as default configuration. I added it to keep consistency for a future. I don't use static actually, but maybe somebody does. What's your suggestion? Remove it?

File: libs/libcoap/Makefile (hunk 4 above)
Comment (toxxin) [inline libs/libcoap/Makefile]: Done. Update as soon as we clarify the issue above.

File: libs/libcoap/Makefile (hunk 3 above)
Comment (mhei) [inline libs/libcoap/Makefile]: +1 for removing, add it later when it becomes necessary

Comment (toxxin) [inline libs/libcoap/Makefile]: Agree. Fixed.

**Comment by karlp** [comment]
Merged in bf45232

**Pull request closed without merging**